socketio/src/payload.rs

```diff
@@ -2,13 +2,15 @@ use bytes::Bytes;
 
 /// A type which represents a `payload` in the `socket.io` context.
 /// A payload could either be of the type `Payload::Binary`, which holds
-/// data in the [`Bytes`] type that represents the payload or of the type
-/// `Payload::String` which holds a [`std::string::String`]. The enum is
+/// data in the [`Bytes`] type that represents the payload, of the type
+/// `Payload::String` which holds a [`std::string::String`] or of the type
+/// `Payload::StringArray` which holds a [`Vec<std::string::String>`]. The enum is
 /// used for both representing data that's send and data that's received.
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub enum Payload {
     Binary(Bytes),
     String(String),
+    StringArray(Vec<String>),
```

Comment:
(reflecting on previous decision to use string and conditionally interoperate as JSON) Is string really the type we want here? From our usage (checking to see if the string is valid JSON) it looks more like we mean serde_json::Value.

We could use an intermediate type with some From conversions to make the ergonomic

```rust
struct TextPayload(serde_json::Value);

enum Payload {
  Binary(Vec<u8>),
  Text(Vec<TextPayload>)
}

impl From<String> for Payload {
  fn from(string: String) -> Self {
    Self::Text(vec![TextPayload::from(string)])
  }
}

impl From<String> for TextPayload {
  fn from(string: String) -> Self {
    if let Ok(value) = serde_json::from_str::<serde_json::Value>(&string) {
        Self(value)
    } else {
        Self(serde_json::Value::String(string))
    }
  }
}
```

Reply:
Supporting multiple arguments seems like a reasonable reason for a minor version bump IMO. Everyone will want this feature

Reply:
this sounds great, and storing the serde value directly will definitly be easier, i'll get working on it when i get some free time

Reply:
but wouldn't there be problems with existing code if the enum variant name and inner value changed ?

Reply:
> but wouldn't there be problems with existing code if the enum variant name and inner value changed ?

Yes, fair point. We should be nice to existing users and deprecate the old one. Updated original post to include the current String variant.

existing `match` expressions on this will also break with the addition of any variants to Payload, which seems unavoidable in order to support this feature (which we should).